### ISSUE TITLE
Somethings look different with the paper.

### ISSUE THREAD

**Issue by Haetsal-Lee** [fabienbaradel/glimpse_clouds#6]
It seems there are few things that are implemented different with papers.

1. Encourage diversity of attention points
```python
dist_btw_glimpses = mse_loss(attention_points_1, attention_points_2)
```
The above value, dist_btw_glimpses is just summed up with the weight alpha_encourage_diversity=1.0.
  This will discourage diversity seems different rather than encouraging. 
Would't it discourage diversity of attention points?
2. About Memory bank, slots
  It seems there is no operation for deleting old slots.
3. D is trained end-to-end, but not saved.
Paper says
> After pre-training, D is trained end-to-end.
On codes, Itt seems that D is trained but not saved. And re-calculated at first batch time whenever model reloaded. So It seems that it is pointless to train D.

**Comment by fabienbaradel** [comment]
Hi @Haetsal-Lee,

First of all thanks for the interest in our work. Below are my answers point by point:
1. I am confused about this point and I think that you are right the function should returned ```1/(1+dist_btw_glimpses)``` instead of returning ```dist_btw_glimpses``` (cf L_G_1 in the paper in Eq. 15).
2. Since we are dealing with short-sequences (8 frames only) we are not deleting the old slots. So the memory slot is limited to K=8 in our case.
3. Indeed I have just checked the checkpoint and it seems that D is not part of the model I am sorry for that. For my experiments I was running experiments on the val/test set right after training without saving the model into a checkpoint so I was using D trained on the training set. I apologize for this issue I will update the checkpoint as soon as possible.

**Comment by hchlhwang** [comment]
Is the issue related with the checkpoint planned to be updated?

**Comment by fabienbaradel** [comment]
Hi @HochulHwang , for the moment I will not be able to update the checkpoint. I keep this issue open. I will be happy if you feel free to solve the issue.

**Comment by phamthephuc** [comment]
Hope anybody can solve this issue soon
